File: projectEuler/34/34.java
// Digit Factorials
// 145 = 1! + 4! + 5! = 1 + 24 + 120 = 145

public class 34 {
    public static void main(String[] args) {
        int sum = 0;
        for (int i = 3; i < 1000000; i++) {
            if (i == sumFactorial(i)) {
                sum += i;
            }
        }
        System.out.println(sum);
    }

    public static int sumFactorial(int n) {
        int sum = 0;
        while (n > 0) {
            sum += factorial(n % 10);
            n /= 10;
        }
        return sum;
    }

    public static int factorial(int n) {
        if (n == 0) {
            return 1;
        }
        return n * factorial(n - 1);
    }
}

File: projectEuler/34/34.py
#Digit Factorials
#145 = 1! + 4! + 5! = 1 + 24 + 120 = 145

def factorial(n):
    if n == 0:
        return 1
    else:
        return n*factorial(n-1)

def sumFactorial(n):
    sum =0
    while n>0:
        sum += factorial(n%10)
        n = n//10
    return sum

def main():
    sum = 0
    for i in range(3,100000):
        if i == sumFactorial(i):
            sum += i
    print(sum)   

if __name__ == "__main__":
    main()
       